lauth/app/ops/authorize.rb

```diff
@@ -0,0 +1,56 @@
+module Lauth
+  module Ops
+    class Authorize
+      include Deps[
+        "repositories.grant_repo",
+        "repositories.collection_repo"
+      ]
+
+      def call(request:)
+        collection = collection_repo.find_by_uri(request.uri)
+        case collection.dlpsAuthzType
+        when "n"
+          normal_mode(request: request)
+        when "d"
+          delegated_mode(request: request, collection: collection)
+        else
+          raise "Unknown dlpsAuthzType '#{collection.dlpsAuthzType}'"
```

Comment:
Probably want a typed error if raising is the right path

Reply:
It should be unreachable but it isn't sufficiently constrained that I didn't want the safeguard in place for future development.